### ISSUE TITLE
Remove/update version-specific code paths

### ISSUE THREAD

**Issue by sophiewigmore** [cloudfoundry/dotnet-core-buildpack#455]
## Background

In the process of removing .NET Core 2.1 from the buildpack, we stumbled across multiple explicit code paths that were in place for .NET Core 1.X and 2.1. We removed some of these ([change 1](https://github.com/cloudfoundry/dotnet-core-buildpack/commit/69739bfa8c95adc0c329a216327e3b09281c03e1) and [change 2](https://github.com/cloudfoundry/dotnet-core-buildpack/commit/f041167a258e340346a4176e77efb1648587a901)).

## Issue

Since this wasn't our prime objective while removing 2.1, we only removed explicit version calls. We discovered less obvious code paths such as the one files in #447 that use out of date .NET Core features. 

In this issue, we should do a buildpack-wide search of the different criteria we use for detection, supply, finalizing, etc and:
* identify out of date logic (for example, what we did in https://github.com/cloudfoundry/dotnet-core-buildpack/commit/69739bfa8c95adc0c329a216327e3b09281c03e1)
* identify new logic we don't support for 3.1 and 5.0 (for example, searching for `FrameworkReference` in #447)

Depending on the difficulty of the change, you may either fix it as a part of this issue, or file a separate issue to tackle it and link to that issue in the comment section of this issue.

It may be useful to reference the official .NET Core version migration guides for this:
https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-5.0&tabs=visual-studio
https://docs.microsoft.com/en-us/aspnet/core/migration/30-to-31?view=aspnetcore-5.0
https://docs.microsoft.com/en-us/aspnet/core/migration/31-to-50?view=aspnetcore-5.0

## ~~Issue blocked on #446~~
~~The completion of this issue will unblock this issue. It's completion will lead to 5.0-specific code paths being added which takes care of a part of the "update" step in this issue.~~

## Acceptance
Acceptance for this issue is tricky since there isn't a nice definitive list of features to check for. For each identified piece of logic for removal or addition, it should be fixed as a part of this issue or a new issue should be filed.

**Comment by johnnyr0x** [comment]
Is this still applicable? @sophiewigmore

**Comment by johnnyr0x** [comment]
Per our discussion this 3rd bullet as well:

  - What context would you share with someone who hasn't seen the issue?
